festim/concentration/traps/extrinsic_trap.py

```diff
@@ -55,6 +56,23 @@ def __init__(
         self.density_previous_solution = None
         self.density_test_function = None
 
+    @property
+    def newton_solver(self):
+        return self._newton_solver
+
+    @newton_solver.setter
+    def newton_solver(self, value):
+        if value is None:
+            self._newton_solver = value
+        elif isinstance(value, NewtonSolver):
+            if self._newton_solver:
+                warnings.warn(
+                    "Settings for the Newton solver will be overwritten", UserWarning
+                )
```

Comment:
I wouldn't make this an actual `UserWarning` but rather a print statement. The reason for this is that users could never get rid of this warning. 

Same for other classes.

```suggestion
                print("Settings for the Newton solver will be overwritten")
```

Reply:
OK